### ISSUE TITLE
Editor: date in sidebar is very confusing

### ISSUE THREAD

**Issue by designsimply** [Automattic/wp-calypso#1763]
Raised by @rralian 

I've been working with the post-scheduler recently, and I find the behavior of the date displayed in the "advanced status" sidebar to be extremely confusing and unclear. We don't really give any indication about what that date means, and we actually change its meaning in different situations without explanation.

### Last-Modified Date
Normally the date displayed is the last-modified date for a draft. I think the intent is to give you reassurance that your Post has saved. Although that assurance is handled with the content above ("draft saved a few seconds ago").

![last modified date](https://cloudup.com/cKm5Au3qfjN+)

### Or Is It Published Date?
If I publish the post, then it becomes the publish date, which is no longer the last-modified date. Here I set the Publish date to a day in the past (whereas last-modified date is today). Again, its purpose is somewhat duplicated with the content above.

![published date](https://cloudup.com/cFxOYr61YJB+)

### Or Is It Scheduled Date?
If I schedule a Post to publish on a future date, the date displayed is now the scheduled publish date. We add a "FUTURE" tag to the date, which... I can confirm that is a date in the future, but that's not telling me what the date actually means. Again it's duplicated above.

![scheduled date](https://cloudup.com/cK8AwH6i-ok+)

In each of these cases, clicking on the date will open a calendar dialog that will allow you to schedule the Post. That really doesn't make sense when clicking the "last-modified" date. It does make sense however to modify the published date or the scheduled date.

### Back to the Future
And I can get myself in fun situations like this. I first scheduled a post in the future. Then I selected a date in the past. While the date is reflecting a date in the past here, it still shows the label "future". _Back to the Future!_

![back to the future](https://cloudup.com/cwt5y2rVeuW+)

Part of the issue leading to that screenshot is that we're updating this date in the sidebar immediately when you click on a calendar date, but we don't update its status. And the post-scheduler behaves differently depending on how you close it. You can open the scheduler, click a date, and then you can close it again by clicking on the calendar icon (toggles it closed), by clicking save (stores the publish date, but does not affect schedule status), or by clicking the "schedule" button (stores the date and updates the status to "scheduled").

## Recommendation
I think the confusion mostly comes from the visual separation of this date from the status in the section above. So the date effectively has no label. I think instead we should place that date immediately underneath the status, not in a separate section. I also think it's unnecessary to display the date when this is just a draft and we're already saying how old the draft is. And I think the badge "FUTURE" is unnecessary for a scheduled Post. So I would suggest this treatment for the different situations. I think this is more clear.

> **DRAFT** SAVED -- A FEW MINUTES AGO  

> **PUBLISHED** -- 2 DAYS AGO
Oct 28, 2015 4:15pm 

> **SCHEDULED** -- IN 1 DAY
Oct 31, 2015 4:15pm 

I also think the scheduler should _not_ immediately update the date shown in the sidebar when you select a day. Instead that should be ephemeral until you click the "schedule/update" button or "save" link, both of which should update the date _and_ the the status. Or if you toggle the calendar closed again (by clicking the calendar icon, the date in the sidebar, or hitting escape), that should just close the calendar and no change should be made.

@kellychoffman @retrofox thoughts?

**Comment by designsimply** [comment]
Copying over a comment from @folletto 

Excellent analysis. After reading it my thought was "The two field should be independent and separate", or in other terms:

* The top bit should always refer to the saved date.
* The bottom date should always be the publish / scheduled date.

But, I also notice that the top part is very effective alone in mixing well both status ("Published") and save status, so the semantics overlap a fair bit. It's also fairly counter-intuitive that once scheduled you have no idea anymore when the last save happened... but maybe it's not super relevant?

The pairing of the full date inside the drawer with the revisions makes me also think that the date showed there could always be the latest revision date, so always save date. Do we have it? Or it just overrides it with the scheduled one?

Sidenote: we can't go on the same line with long text afaik: "DRAFT SAVED -- A FEW MINUTES AGO" won't fit considering also i18n and the "Save" / "Saving" badge on the right. That's why we went on two rows there.

I'm afraid I just added even more trouble... I don't have a clear solution. However, I entirely agree on the issue and the problem you raise.

Iterating on your proposal:

> DRAFT SAVED 
> A FEW MINUTES AGO

> PUBLISHED
> 2 DAYS AGO AT 4:15pm

> PUBLISHED
> OCT 28, 2015 4:15pm

> SCHEDULED
> FOR OCT 31, 2015 4:15pm

* The "delta" date shouldn't be used imho for scheduled and times beyond just a few days, make it harder to position in time imho.
* The date below the folded panel then would always be the "Last Save" date.

**Comment by designsimply** [comment]
Copying over a clarification from @rralian 

Maybe this screenshot can better clarify the issue I'm trying to address. I scheduled the post to publish in the future. And then I opened the calendar and selected a date in the past.

![confusing state](https://cloudup.com/c25-tc-jc9Y+)

Notice that we updated the date in the sidebar, but we didn't change anything about the status (still showing "scheduled" and "in a day"). So right there the UX is contradicting itself. And the date is in the past, but we have the badge "future" on it. That doesn't make any sense.

I can now close the calendar without saving it just by clicking on the calendar icon, and this is how it will remain in the page.

![another weird state](https://cloudup.com/cN5wkR706n4+)

It seems like I've made a change here but I have not. It's just reflecting some weird contradicting state. If I reload the page, this is what I see instead because nothing has been saved.

![what is actually saved](https://cloudup.com/c3-SNrYs2jo+)

We can avoid this confusion by treating the calendar as a two-step process. Click a day (which is _only_ reflected in the calendar UX), and then click "schedule/update" to confirm that new date (and then see it reflected in the status/date in the sidebar). In fact we are already forcing this two-step process anyway... because if you click a day and then do not click "schedule/update", then the app breaks in interesting ways.

Reply from @folletto 

> Agreed. The update of the date without an actual "save" is misleading. It would be still a bit "cryptic" that you actually have to click Update after clicking the date but would be better already.
